src/app/shared/course-form/course-form.component.html

```diff
@@ -1,50 +1,50 @@
-<form (ngSubmit)="onSubmit($event)" class="course-form-wrapper" *ngIf="course">
+<form class="course-form-wrapper" *ngIf="course" [formGroup]="courseForm" (ngSubmit)="onSubmit()">
     <div class="label-wrapper">
         <label class="input-label" for="title">Title</label>
-        <span class="required">*required</span>
+        <span class="required" *ngIf="courseForm.get('name').touched && courseForm.get('name').invalid">Required Field, Max length 50 characters</span>
     </div>
-    <input [(ngModel)]="course.name" class="course-input course-input-large" id="title" name="title"
+    <input [formControlName]="'name'" class="course-input course-input-large" id="title" name="title"
```

Comment:
you could use
<input formControlName="name" instead of [formControlName]="'name'"